s3fs/core.py

```diff
@@ -1296,6 +1311,8 @@ def readline(self, length=-1):
 
         If length is specified, at most size bytes will be read.
         """
+        if self.file_backed_cache:
+            raise ValueError('readline not available in file backed cache mode')
```

Comment:
I notice that the mmap API supports readline directly https://docs.python.org/3/library/mmap.html#mmap.mmap.readline